src/v/io/tests/persistence_test.cc

```diff
@@ -0,0 +1,302 @@
+/*
+ * Copyright 2023 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+#include "io/persistence.h"
+#include "io/tests/common.h"
+#include "test_utils/test.h"
+#include "units.h"
+
+#include <seastar/util/defer.hh>
+#include <seastar/util/file.hh>
+#include <seastar/util/tmp_file.hh>
+
+namespace io = experimental::io;
+
+template<typename T>
+class PersistenceTest : public ::testing::Test {
+public:
+    void SetUp() override {
+        dir = seastar::make_tmp_dir(".").get();
+        fn = make_filename();
+    }
+
+    void TearDown() override {
+        for (auto& file : open_files) {
+            file->close().get();
+        }
+        dir.remove().get();
+    }
+
+    std::filesystem::path make_filename() {
+        return dir.get_path() / fmt::format("file.{}", count++);
+    }
+
+    // fs::create wrapper that records file to close after test finishes
+    auto create(std::filesystem::path path) {
+        auto f = this->fs.create(path).get();
+        open_files.push_back(f);
+        return f;
+    }
+
+    // fs::open wrapper that records file to close after test finishes
+    auto open(std::filesystem::path path) {
+        auto f = this->fs.open(path).get();
+        open_files.push_back(f);
+        return f;
+    }
+
+    T fs;
+    seastar::tmp_dir dir;
+    std::filesystem::path fn;
```

Comment:
What is `fn`? I'm familiar with this being short for `function`. Should this be `path` to make the tests more readable?

Reply:
`fn` = `filename`. i changed to `path` to avoid confusion.